superset/assets/src/visualizations/deckgl/DeckGLContainer.jsx

```diff
@@ -53,7 +55,9 @@ export default class DeckGLContainer extends React.Component {
   }
   tick() {
     // Limiting updating viewport controls through Redux at most 1*sec
-    if (this.state && this.state.previousViewport !== this.props.viewport) {
+    // Deep compare is needed as shallow equality doesn't work here, viewport object
+    // changes id at every change
```

Comment:
nit:
```
const { previousViewPort } = this.state;
const { viewport } = this.props;
if (this.state && !isEqual(previousViewport, viewport)) {
   ...
}
```

Reply:
Careful here though `const { previousViewPort } = this.state;` will fail if `this.state` is `undefined`

Reply:
isn't the state always going to be defined with this line:
https://github.com/apache/incubator-superset/pull/6363/files/052fe92b28888a5d428575a0dde50bd4a7900388#diff-52ae534f640f8c5b7eb6e05bd0e56259R30